mzLib/UsefulProteomicsDatabases/ProteinDbWriter.cs

```diff
@@ -0,0 +1,136 @@
+﻿using Proteomics;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Linq;
+using System.Globalization;
+
+namespace UsefulProteomicsDatabases
+{
+    public class ProteinDbWriter
+    {
+        #region Public Methods
+
+        public static void WriteXmlDatabase(Dictionary<string, HashSet<Tuple<int, ModificationWithMass>>> Mods, List<Protein> proteinList, string outputFileName)
+        {
+            var xmlWriterSettings = new XmlWriterSettings
+            {
+                Indent = true,
+                IndentChars = "  "
+            };
+
+            using (XmlWriter writer = XmlWriter.Create(outputFileName, xmlWriterSettings))
+            {
+                writer.WriteStartDocument();
+                writer.WriteStartElement("uniprot");
```

Comment:
Another way of thinking about this is that this is the name of the entity that generated the database. Let's change this to "mzLib" by default, and allow passing a string parameter that stands for the original source database, that would be concatenated with mzLib. 
So it could be something like:
"mzLib"  or "mzLib and uniprot"  or "mzLib and SomeOtherSource". 
We don't even know if we are using the real uniprot xml structure: we're not using an xsd file, and just writing stuff to a file that looks kindof like uniprot xml